modules/custom/moj_resources/src/Utilities.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Drupal\moj_resources;
+
+/**
+ * Utilities
+ */
+
+class Utilities
+{
+  public static function filterByPrisonTypes($prison_types, $query) {
+    $prison_type_or_group = $query
+      ->orConditionGroup()
+      ->condition('field_prison_types', $prison_types, 'IN')
+      ->condition('field_prison_types', NULL, 'IS NULL')
+      ->notExists('field_prison_types');
```

Comment:
Does this match the logic in https://docs.google.com/document/d/1z2A_NxAy-dJ6_SmByUnSXIq2WKvmQxNqodZNbG0ZnNo/edit? It looks like it includes an item when the field is empty on L15/16), but in that case I'd expect that item to be excluded from the set. I might be reading this wrong though.

Reply:
IIRC a prison type can be `null` for a piece of content, we already validate that the series has a matching prison type against the prison before executing the query

Reply:
:+1: for an episode it may be `null` if the series has the prison type set. Maybe the name of this function is what's off, as the logic for filtering by prison type is (normally) exclusive of `null`? 

This makes me wonder:
- Does this want to live in a utility function if it only applies to episodes in a series? 
- Thinking about making the business logic cohesive, should the function be aware of the series/episode relationship?

Reply:
But assuming no `prison_type` would be implicitly `global` would that mean this filter is valid for our use case?

Reply:
From [Side note: making things less implicit](https://docs.google.com/document/d/1z2A_NxAy-dJ6_SmByUnSXIq2WKvmQxNqodZNbG0ZnNo/edit), if no `prison_type` is present, that means "no prison types" rather than "all prison types".